### PR TITLE
Fixes and changes to support compiler server

### PR DESCRIPTION
**Pull request by kg**
This PR contains some bug fixes and infrastructure changes to support use of the compiler server.

### CODE REVIEW

File: mcs/class/System.Core/build_System.Core.dll.sources [hunk 1]
```diff
@@ -1,2 +1 @@
-#include net_4_x_System.Core.dll.sources
-#include pipes_pns.sources
+#include unix_net_4_x_System.Core.dll.sources
```
Comment (marek-safar) [inline mcs/class/System.Core/build_System.Core.dll.sources]: Shouldn't this be in unix_build_System.Core.dll.sources only?

Comment (kg) [reply mcs/class/System.Core/build_System.Core.dll.sources]: You're right, I'll fix it.

**Comment by kg** [comment]
@monojenkins build Windows x64

**Comment by kg** [comment]
@monojenkins build failed

**Comment by marek-safar** [comment]
@kg the failing tests are related

**Comment by kg** [comment]
I've been looking at the mac failures, the windows ones were the basic-profile-check issue and the wasm ones were a wasm provisioning problem.

**Comment by filipnavara** [comment]
Thanks for taking care of it!

**Comment by kg** [comment]
@monojenkins build failed

**Comment by kg** [comment]
@monojenkins commit csproj

**Comment by kg** [comment]
@monojenkins build failed

**Pull request merged**